.changeset/tidy-seas-sing.md

```diff
@@ -0,0 +1,79 @@
+---
+"thirdweb": patch
+---
+
+Added Bridge.Onramp.prepare and Bridge.Onramp.status functions
+
+## Bridge.Onramp.prepare
+
+Prepares an onramp transaction, returning a link from the specified provider to onramp to the specified token.
+
+```typescript
+import { Bridge } from "thirdweb";
+import { ethereum } from "thirdweb/chains";
+import { NATIVE_TOKEN_ADDRESS, toWei } from "thirdweb/utils";
+
+const preparedOnramp = await Bridge.Onramp.prepare({
+  client: thirdwebClient,
+  onramp: "stripe",
+  chainId: ethereum.id,
+  tokenAddress: NATIVE_TOKEN_ADDRESS,
+  receiver: "0x...", // receiver's address
+  amount: toWei("10"), // 10 of the destination token
+  // Optional params:
+  // sender: "0x...", // sender's address
+  // onrampTokenAddress: NATIVE_TOKEN_ADDRESS, // token to initially onramp to
+  // onrampChainId: 1, // chain to initially onramp to
+  // currency: "USD",
+  // maxSteps: 2,
+  // purchaseData: { customId: "123" }
```

Comment:
We should include an optional country param in case calling from the server

Reply:
btw i dont think the ip country detection works, its letting me see stripe stuff which is not available in NZ

Reply:
I don't think it does either and I don't know why